src/Frame.js

```diff
@@ -0,0 +1,59 @@
+function Frame(frameNum) {
+  this.rollCounter = 0;
+  this.rolls = [];
+
+  this.strike = false;
+  this.spare = false;
+
+  this.currentScore = 0;
+  this.frameNum = frameNum;
+}
+
+Frame.prototype.roll = function roll(pinsKnocked) {
+  this.isRollLegit();
+  this.isStrike(pinsKnocked);
+
+  if (this.strike !== true) {
+    this.rolls.push({ roll: parseInt(`${pinsKnocked}`, 10) });
+    this.calculateScore(pinsKnocked);
+    this.isSpare();
+  }
+
+  return pinsKnocked;
+};
+
+// PRIVATE METHODS
+
+Frame.prototype.isRollLegit = function isRollLegit() {
+  const newCounter = this.rollCounter + 1;
+
+  if (this.strike === false && this.spare === false && this.frameNum === 10 && newCounter === 3) {
+    throw new Error('The game is over!');
+  } else if ((this.strike || this.spare) === true && this.frameNum === 10 && newCounter === 4) {
+    throw new Error('The game is over!');
+  } else if (this.strike === true && this.frameNum === undefined) {
+    throw new Error('You scored a strike. Start next frame!');
+  } else if (newCounter >= 3 && this.frameNum === undefined) {
+    throw new Error('You cannot roll again. Start next frame!');
+  }
+
+  this.rollCounter += 1;
+};
+
+Frame.prototype.isStrike = function isStrike(pinsKnocked) {
```

Comment:
This function feels to me like it has multiple responsibilities at the moment.  an isStrike function sounds like it should just check whether a particular roll is a strike, this seems to do that, then calculate the score, then push the rolls into storage.  Could you perhaps break some of this functionality out into it's own function in order to keep to the single responsibility principle.